python/pylibvw.cc

```diff
@@ -54,6 +57,113 @@ const size_t pDECISION_SCORES = 8;
 
 void dont_delete_me(void* arg) {}
 
+class OptionManager
+{
+  std::map<std::string, std::vector<VW::config::option_group_definition>> m_option_group_dic;
+  py::object m_cla;
+  VW::config::options_i& m_opt;
+
+public:
+  OptionManager(VW::config::options_i& options, py::object cla)
+      : m_opt(options), m_option_group_dic(options.get_collection_of_options()), m_cla(cla)
+  {
+  }
+
+  template <typename T>
+  py::object* value_to_pyobject(VW::config::typed_option<T>& opt)
+  {
+    return new py::object(
+        m_cla(opt.m_name, opt.m_help, opt.m_short_name, opt.m_keep, opt.m_necessary, opt.m_allow_override, opt.value(),
+            m_opt.was_supplied(opt.m_name), opt.default_value(), opt.default_value_supplied()));
+  }
+
+  template <typename T>
+  py::object* value_to_pyobject(VW::config::typed_option<std::vector<T>>& opt)
+  {
+    py::list values;
+
+    auto vec = opt.value();
+    if (vec.size() > 0)
+    {
+      for (auto const& opt : vec) { values.append(py::object(opt)); }
+    }
+
+    return new py::object(m_cla(opt.m_name, opt.m_help, opt.m_short_name, opt.m_keep, opt.m_necessary,
+        opt.m_allow_override, values, m_opt.was_supplied(opt.m_name), py::list(), opt.default_value_supplied()));
+  }
+
+  template <typename T>
+  py::object* transform_if_t(VW::config::base_option& base_option)
+  {
+    if (base_option.m_type_hash == typeid(T).hash_code())
+    {
+      auto typed = dynamic_cast<VW::config::typed_option<T>&>(base_option);
+      return value_to_pyobject(typed);
+    }
+
+    return nullptr;
+  }
+
+  template <typename TTypes>
+  py::object base_option_to_pyobject(VW::config::base_option& options)
+  {
+    py::object* temp = transform_if_t<typename TTypes::head>(options);
+    if (temp != nullptr)
+    {
+      auto repack = py::object(*temp);
+      delete temp;
+      return repack;
+    }
+
+    return base_option_to_pyobject<typename TTypes::tail>(options);
+  }
+
+  py::object get_vw_option_pyobjects(vw_ptr all, bool enabled_only)
+  {
+    py::dict dres;
+    auto it = m_option_group_dic.begin();
+
+    auto enabled_reductions = all->enabled_reductions;
+
+    while (it != m_option_group_dic.end())
+    {
+      auto reduction_enabled =
+          std::find(enabled_reductions.begin(), enabled_reductions.end(), it->first) != enabled_reductions.end();
+
+      if (((it->first).compare("general") != 0) && enabled_only && !reduction_enabled)
```

Comment:
same here, is there a way to hide the "general" string?

Reply:
done